CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -5,6 +5,7 @@
 #include <algorithm>
 
 #include "CUDADataFormats/Track/interface/TrajectoryStateSoAT.h"
+#include "Geometry/TrackerGeometryBuilder/interface/phase1PixelTopology.h"
```

Comment:
how strict are the rules for the dependencies in the CUDADataFormats?
@makortel please clarify.

Here, a dependence on `Geometry/TrackerGeometryBuilder` seems somewhat out of order.

Reply:
> how strict are the rules for the dependencies in the CUDADataFormats?

The `CUDADataFormats` fall in the same category as "transient products".

> Here, a dependence on `Geometry/TrackerGeometryBuilder` seems somewhat out of order.

There is already a dependence here
https://github.com/cms-sw/cmssw/blob/df8ebddd58a10162ac348d34b858a8adca8cd824/CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DSOAView.h#L9
so this PR is not making things (much) worse.

But probably (after this PR) it would be necessary to think again its placement, because in the long term I'd assume classes using the `phase1PixelTopology` to be persisted (at which point the relevant classes should be defined in `DataFormats`).

Reply:
Dependency on Geometry is allowed in DataFormat since ages

Reply:
`DataFormats` are allowed to depend on some subpackages of `Geometry`, but not everything. Currently the set is limited to
* `Geometry/CaloGeometry`
* `Geometry/CommonDetUnit`
* `Geometry/CommonTopologies`
* `Geometry/GEMGeometry`

(from FWLite build set). `Geometry/TrackerGeometryBuilder` itself would bring in
* `DetectorDescription/Core`
* `DetectorDescription/DDCMS`
* `CondFormats/GeometryObjects`
* `FWCore/ParameterSet` (although this might be possible to remove?)
* `dd4hep`

These dependencies would be too much for `DataFormats` (although there is already one case depending on `FWCore/ParameterSet`).

Reply:
for what I am concerned ```phase1PixelTopology.h``` can move anywhere else: the whole idea is to have geometry constants as constant not as database or DD quantities!

Moving elsewhere will need changes in many packages. I propose to open an issue and not hold this specific PR.

Reply:
I opened an issue https://github.com/cms-sw/cmssw/issues/36302